lib/Graph/Graph.cpp

```diff
@@ -421,22 +421,29 @@ Placeholder *Module::addPlaceholder(Placeholder *ph) {
   return ph;
 }
 
+/// Check the kernel size for Conv/Pooling ops.
+static void checkKernelSize(ShapeNHWC idim, llvm::ArrayRef<unsigned_t> kernels,
+                            llvm::ArrayRef<unsigned_t> pads) {
+  PaddingTLBR pdim(pads);
+  (void)pdim;
+  ShapeHW kdim(kernels);
+  (void)kdim;
+  assert((idim.w + pdim.left + pdim.right) >= kdim.width &&
+         (idim.h + pdim.top + pdim.bottom) >= kdim.height &&
+         "Kernel size is too large");
+}
```

Comment:
These checks already happen in Convolution::verify.
I suggest to call the verify (wrapped into an assert) instead of duplicating the logic.
One way to easily fix that would be to call ::verify on every node that we add to the graph and make sure all the checks we have here are in the verifier.
I'll file an issue in that direction.